lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -218,6 +269,8 @@ public function execute()
      * </code>
      *
      * @return string The SQL query string.
+     *
+     * @throws QueryException If the SELECT query is empty.
```

Comment:
The "SELECT query is empty" reason will be reflected in the exception message. I'd keep it generic like “If the object doesn't represent a valid query in its current state”.

Reply:
Done.